// vue-recaptcha/dist/nuxt.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
module.exports = void 0;
var _kit = require("@nuxt/kit");
const COMPONENTS = {
  RecaptchaCheckbox: "Checkbox",
  RecaptchaChallengeV2: "ChallengeV2",
  RecaptchaChallengeV3: "ChallengeV3"
};
var _default = (0, _kit.defineNuxtModule)({
  meta: {
    name: "vue-recaptcha",
    compatibility: {
      nuxt: "^3.0.0"
    }
  },
  async setup(opt) {
    const resolver = (0, _kit.createResolver)(require('url').pathToFileURL(__filename).toString());
    for (const [name, fileName] of Object.entries(COMPONENTS)) {
      await (0, _kit.addComponent)({
        name,
        global: opt._globalComponent,
        filePath: resolver.resolve(`./components/${fileName}.vue`)
      });
    }
    (0, _kit.addImports)([{
      from: resolver.resolve("./composables/context"),
      name: "useRecaptchaContext"
    }, {
      from: resolver.resolve("./composables/context"),
      name: "useRecaptchaProxy"
    }, {
      from: resolver.resolve("./composables/challenge-v2"),
      name: "useChallengeV2"
    }, {
      from: resolver.resolve("./composables/challenge-v3"),
      name: "useChallengeV3"
    }, {
      from: resolver.resolve("./composables/script-provider"),
      name: "useRecaptchaProvider"
    }]);
    (0, _kit.addPlugin)({
      src: resolver.resolve("./nuxt-plugin")
    });
  }
});
module.exports = _default;
